### ISSUE TITLE
Order Creation: Accessibility → Product rows in order should be marked as button

### ISSUE THREAD

**Issue by rachelmcr** [woocommerce/woocommerce-ios#6125]
In order creation, selecting a product in the order opens a Product detail screen (e.g. to remove the product from the order).

When using Voiceover, we should mark each product row as a button with a hint like "View or remove this product from the order".

<img src="https://user-images.githubusercontent.com/8658164/153413903-334bc1c1-a737-4902-ad36-edeb28b36f97.PNG" width="300px">
